cpp/README.md

```diff
@@ -42,3 +42,41 @@ See https://jrhemstad.github.io/nvtx_wrappers/html/index.html for Doxygen docume
   }
   ```
 
+  ## Getting NVTX++
+
+  The NVTX C++ wrappers are header-only and implemented in a single header file.
+  This header can be incorporated manually into your project by downloading the header and placing it in your source tree. 
+  However, we strongly recommend using CMake or CMake Package Manager (CPM) to fetch and link NVTX++ to your project.
```

Comment:
I don't like the "strongly" here -- it makes it sound like there's something wrong with just copying the files into your codebase.  Not everyone uses CMake -- I heard that a few times at the hackathon last week.  And not everyone is OK with requiring network access from their build (for example all NVIDIA sw that builds in DVS), so I want this part to start by saying e.g."Here are three easy ways to get NVTX for C++:  (1) Download a copy of the headers, place them in your code, and make sure to add the necessary -I to your compile command line or project settings  (2) Download a copy of the headers and use them from CMake  (3) Use CMake and CPM to automatically download NVTX from github in your build".  And list the pros & cons of each, since they're all valid use cases.  That way people see there's nothing fancy going on.  Your comments here are great for (3).

Reply:
>it makes it sound like there's something wrong with just copying the files into your codebase.

But that method is strictly worse. It requires the user to add the `--std=c++11` option to their compile flags, it requires them to make sure the NVTX C headers are in the include path, and it requires them to link against `libdl`. All of these things are encapsulated by the CMake target. 

>  And not everyone is OK with requiring network access from their build

The way CPM works, it will look for a package locally before trying to download it.

Reply:
For people whose build is currently Makefiles or IDE-specific projects or hand-written shell scripts, coercing them to change to CMake is not the most inviting approach.  I know it sucks, but this is a large number of people still -- e.g. the CUDA SDK samples ship with Makefiles and VS projects.  I certainly agree with you that CMake is cool and CPM is even cooler, and I am grateful you put in the time to get these working here!  It's a very elegant solution for CMake users.  But I don't want NVTX to make people think they need these things to use NVTX.  The whole point of NVTX is to work everywhere, dependency-free, and that includes the build itself.  Everything since GCC 6 and MSVC 2015 default to at least C++14, so only GCC 4/5 users have to specify `-std=c++11`, and that's 10 characters in the Makefile.  People using Makefiles or other build systems will know how to add an include path and to make sure `-ldl` gets added on Linux.  Since this is all standard fare for a C/C++ library, I want NVTX for C and C++ to make it super clear that these typical things are all you need to do to use it.  We don't even make you link against static or shared libs.  The old-school folks will not find this difficult.  But yeah, then I love that we can say "if you're using CMake, just paste in this blurb and it will do all those things for you!".

Also, about "CPM will look for a package locally before trying to download it", can we add to this text how to tell it where to look locally?  Basically if someone has already downloaded NVTX C or C++ and checked it into source control, and have to use that checked-in version, we should show them how to point to their local copy.  CPM is super cool, but some teams (like mine for example) require all dependencies be checked into source control, no exceptions -- I want these folks to see there's a simple solution for them too.  This is why I'd like the doc to present those three options:  (1) the bare-metal manual way for any build system, (2) the CMake purely-from-source-control way, and (3) the CPM way.